### ISSUE TITLE
Upgrade to newer Bitbucket API and remove deprecated classes

### ISSUE THREAD

**Issue by christiangalsterer** [christiangalsterer/stash-filehooks-plugin#37]

**Comment by raspy** [comment]
I would postpone implementing this as long as line 4.x did not reach end of life. There is no point in dropping 4.x line support if possible, especially since this line is probably still widely deployed in enterprises, where it's not that easy to upgrade.

**Comment by hogarth-sv** [comment]
Is this the reason that #52 is an issue?

Feels like a 4.0.0 release dropping the 4.x support now that 6.x is out makes sense then ...

**Comment by ar613** [comment]
Yes, the plugin needs significant rework to use the new plugin API in Bitbucket 6. Is the project still alive? Last commit was almost a year ago.
